app/models/base_facility.rb

```diff
@@ -153,6 +161,18 @@ def build_result_set(bbox_num, type, services)
       TYPES.map { |facility_type| get_facility_data(conditions, type, facility_type, services) }.flatten
     end
 
+    def build_result_set_from_ids(ids)
+      ids_for_types = ids.split(',').each_with_object({}) do |type_id, obj|
+        facility_type, unique_id = type_id.split('_')
```

Comment:
Is there much value in building up this dict by facility_type and then iterating by facility_type, vs. just iterating through the raw list of ids and splitting out the prefix each time?

Reply:
I'm not sure. It made more sense to me to build the hash, and use it to make a few `id IN (list_of_ids)` queries, while I was writing it. If we split each time don't we end up with a several `id = vc_0617V` queries?

Reply:
You're right, that's a good rationale

Reply:
But my example of `?ids=foobar` was meant to question the logic on line 167 - if _that_ split returns one object, it will have `facility_type==foobar, unique_id==nil`, there should be a spec around that.

Reply:
Got it. Seems like that's an issue with `find_facility_by_id` too. Would it make more sense to you to ignore a bad id like its one that's not in the database or raise a bad request error?

Reply:
Will do. I wrote a spec and it's definitely a 500 error at the moment (constantize fails). So I'll fix in both places.

Reply:
should be fixed now